loader/src/main/java/net/neoforged/fml/loading/ModDirTransformerDiscoverer.java

```diff
@@ -93,10 +95,10 @@ private static void visitFile(Path path) {
         if (!path.toString().endsWith(".jar")) return;
         if (LamdbaExceptionUtils.uncheck(() -> Files.size(path)) == 0) return;
 
-        SecureJar jar = SecureJar.from(path);
-        jar.moduleDataProvider().descriptor().provides().stream()
-            .map(ModuleDescriptor.Provides::service)
-            .filter(SERVICES::contains)
+        JarMetadata metadata = JarMetadata.from(new JarContentsBuilder().paths(path).build());
+        metadata.providers().stream()
```

Comment:
This greatly speeds up scanning for providers because we are not computing the packages anymore.